app/src/main/kotlin/com/hedvig/app/feature/claims/ui/commonclaim/CommonClaimActivity.kt

```diff
@@ -65,7 +66,12 @@ class CommonClaimActivity : AppCompatActivity(R.layout.activity_common_claim) {
 
       firstMessage.commonClaimFirstMessage.text = data.layoutTitle
       firstMessage.commonClaimCreateClaimButton.text = data.buttonText
-      if (data.eligibleToClaim) {
+      if (data.isFirstVet()) {
+        firstMessage.commonClaimCreateClaimButton.enable()
+        firstMessage.commonClaimCreateClaimButton.setHapticClickListener {
+          startActivity(getFirstVetIntent(this@CommonClaimActivity))
+        }
+      } else if (data.eligibleToClaim) {
```

Comment:
This implies that isFirstVet() takes precedence over `eligibleToClaim`. Maybe they should never happen at the same time, but if they do, are we OK with not showing the eligible to claim related content? I guess yes since the only way they will end up here is by pressing on the common claim we'll add for pet specifically?

Reply:
Yes so this basically says: 
- for firstvet - go to their app
- for claims which are eligible to claim - open claims flow
- else disable button